### PR TITLE
Add support on Radio component of multiple values

### PR DESCRIPTION
**Pull request by Mte90**
I was trying to add a radio multiple option but lifter deson't handle them.
This is what happen with my php code without css changed

screenshot_20180222_154046

This with css changed
screenshot_20180222_154816

### CODE REVIEW

**Comment by thomasplevy** [comment]
Beautiful! I need to run a few local tests before merging it in but this should get added in the next few days or early next week at the latest.

Thanks <3

**Comment by Mte90** [comment]
You are welcome :-)
I need that to improve the checkout with more fields :-)

File: includes/llms.functions.core.php [hunk 1]
```diff
@@ -483,10 +483,15 @@ function llms_form_field( $field = array(), $echo = true ) {
 			break;
 
 		case 'checkbox':
-		case 'radio':
 			$checked = ( true === $field['selected'] ) ? ' checked="checked"' : '';
 			$r .= '<input class="llms-field-input' . $field['classes'] . '" id="' . $field['id'] . '" type="' . $field['type'] . '"' . $checked . $disabled_attr . $name_attr . $required_attr . $value_attr . $field['style'] . '>';
 			$r .= $label;
+
```
Comment (thomasplevy) [inline includes/llms.functions.core.php]: Having no break between checkbox and radio means that checkboxes end up having duplicate labels...

Enabling a T&C page via LifterLMS settings will show this issue:

purchase_ _llms-72_dev

File: assets/css/lifterlms.css [hunk 2]
```diff
@@ -2379,7 +2379,7 @@ ul.llms-payment-gateways .llms-payment-gateway-error {
   padding: 0;
 }
 .llms-form-field.type-radio input,
-.llms-form-field.type-radio label, .llms-form-field.type-checkbox input,
+.llms-form-field.type-radio label.llms-radio, .llms-form-field.type-checkbox input,
```
Comment (thomasplevy) [inline assets/css/lifterlms.css]: CSS changes made directly to the compiled `.css` files in the assets directory will be overwritten during build steps.

Instead, make css changes to the .scss file found here: https://github.com/gocodebox/lifterlms/blob/master/_private/scss/_includes/_llms-form-field.scss

After updating you can rebuild the compiled css files via `gulp dist`

See [installing for development](https://github.com/gocodebox/lifterlms#installing-for-development) in the readme.

Comment (thomasplevy) [reply assets/css/lifterlms.css]: As an aside I understand that our SCSS is a little bit of a cluster and I was going to merge this in manually but I found the duplicated checkbox issue while testing

Comment (Mte90) [reply assets/css/lifterlms.css]: you know better then me the code :-)

File: includes/llms.functions.core.php (hunk 1 above)
Comment (Mte90) [reply includes/llms.functions.core.php]: argh I forgot that

File: assets/css/lifterlms.css (hunk 2 above)
Comment (thomasplevy) [reply assets/css/lifterlms.css]: If you can figure out the checkbox issue reliably and want me to deal with the SCSS I can manage :-D

Comment (Mte90) [reply assets/css/lifterlms.css]: added the break, for scss you know better me what to do. I don't have time now to work on that.

**Comment by Mte90** [comment]
any updates for this pr?

**Comment by thomasplevy** [comment]
@Mte90 I haven't had time to rewrite the SCSS and we haven't prioritized the development. If you can write the SCSS and add to the PR it would speed up the process. Thanks

**Comment by Mte90** [comment]
As you asked update the pr with the scss :-)

**Comment by thomasplevy** [comment]
@Mte90 can you send me a code snippet that you're using to test this code PR? Since the core doesn't actually have any radios in any of our fields I've been trying to write a custom field but any way I can come up with to use this is resulting in non-functional radios.

If you could post a working example I can use that to make sure this is all functional before I merge it in and deploy.

Thanks

**Comment by Mte90** [comment]
```function skills_show_backend_user_cf($fields) {
    $fields['pvtazd'] = array(
		'columns' => 12,
		'id' => 'pvtazd',
		'label' => __( 'Privato o Azienda?', 'skills' ),
		'last_column' => false,
 		'options' => array('privato'=>'Privato','azienda'=>'Azienda'),
		'required' => true,
		'type'  => 'radio',
		'value' =>  get_user_meta( get_current_user_id(), 'pvtazd', true)
	);
	llms_cf_forms();
    return $fields;
}
add_filter( 'lifterlms_get_user_custom_fields', 'skills_show_backend_user_cf', 99999);
```

This is enough?

**Comment by thomasplevy** [comment]
@Mte90 that code would only run on the admin panel where the function in question -- `llms_form_field()` -- doesn't ever actually run... is this the right code snippet you're currently using?

**Comment by Mte90** [comment]
You are right, I copied the wrong one and the in the code was written that was for the backend too -.-'
This is the right one:

```
function skills_italian_fisco_regedit_fields( $fields, $screen ) {
	$fields[] = array(
		'columns' => 12,
		'id' => 'pvtazd',
		'label' => __( 'Privato o Azienda?', 'skills' ),
		'last_column' => false,
 		'options' => array('privato'=>'Privato','azienda'=>'Azienda'),
		'required' => true,
		'type'  => 'radio',
		'value' =>  get_user_meta( get_current_user_id(), 'pvtazd', true)
	);
	llms_cf_forms();
	return $fields;
}
add_filter( 'lifterlms_get_person_fields', 'skills_italian_fisco_regedit_fields', 777, 2 );
```

**Comment by Mte90** [comment]
ping :-)

**Comment by Mte90** [comment]
another ping!

**Comment by thomasplevy** [comment]
@Mte90 I'll try to find some availability to run tests against this PR again later this week.

**Comment by Mte90** [comment]
I should have fixed the issues that you faced :-)

**Comment by thomasplevy** [comment]
@Mte90 Please check PHPCS violations identified by Travis

```
FILE: ...ravis/build/gocodebox/lifterlms/includes/llms.functions.core.php
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 571 | ERROR | [x] When an array uses associative keys, each value
     |       |     should start on a new line.
 573 | ERROR | [x] Whitespace found at end of line
----------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

**Comment by thomasplevy** [comment]
@Mte90 this is still not working correctly for me...

I'm still seeing 2 of 3 issues I reported earlier:

1. The radio elements don't "fill" when selected
2. Pre-existing radio elements from the LifterLMS Core (payment gateway selection during checkout) no longer functions with this PR
3. ~Google throws an error due to the presence of duplicate IDs. The iterator loop should append possible the option's value to the end of the ID or something~

Here's a screencap of what I'm seeing now with this code:

purchase_ _llms-72_test

Compared to the existing code in the core:

purchase_ _llms-72_test

I know ya'll are currently using this code in production so I guess it's not causing terrible issues on your site but I'm wondering if that's because there's other custom code already running as well as custom CSS of some kind...

Would you please test this with a theme like twentyseventeen and no other custom code and make sure it all works and provide me with some screenshots of what it looks like!

**Comment by Mte90** [comment]
We are not using this code in production anymore because of the various lifterlms updates.
So I don't know if still works and I have no time to work on that :-/

**Pull request closed without merging**